pkg/dockerregistry/server/app.go

```diff
@@ -55,16 +57,44 @@ type App struct {
 
 	// cache is a shared cache of digests and descriptors.
 	cache cache.DigestCache
+
+	// repositoryEnumerator allows to enumerate repository names known to registry
+	repositoryEnumerator RepositoryEnumerator
+}
+
+type registry struct {
```

Comment:
Note to myself: Maybe worth its own file (to complement already existing `registry_test.go`).